### PR TITLE
fix: @metamask/swaps-controller v6 -> v9

### PR DESCRIPTION
**Pull request by legobeat**
## **Description**

- Upgrade `@metamask/swaps-controller` from v8 to v9
  - Remove redundant patches 

Resolves a bunch of outstanding legacy, including `web-provider-engine`.

Also enables Base Network for Swaps.

## **Related issues**

- #9286 `@metamask/swaps-controller`
  - v8 upgrade 
- #9612
  - reverted v8 upgrade 
- https://github.com/MetaMask/swaps-controller/pull/241
  - fixed issue that motivated revert of v8 upgrade 

#### Blocked by
- [x] https://github.com/MetaMask/swaps-controller/pull/245
  - [x] https://github.com/MetaMask/swaps-controller/pull/219

This PR also patches `swaps-controller@9.0.0` to fix the `web3` import that works differently between the test environment and the actual mobile app / E2E test environment. This should be removed once a `swaps-controller` version addressing this is published. (https://github.com/MetaMask/swaps-controller/pull/255)

## **Manual testing steps**

Regression over swaps functionality in the 4 different scenarios:

ETH -> WETH
WETH -> ETH
ETH -> ERC20
ERC20 -> ETH

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by tommasini** [comment]
Can we take a look at the security bot issues?
Also if it's possible can we had a recording of a swap working on the PR?

**Review by tommasini — APPROVED** [review]
Overall code LGTM! Just the socket security bot flags need to be investigated! Also if possible can we have a recording on the PR of swaps working?
Manual steps included are also appreciated for QA on the release dutty!

**Comment by nikoferro** [comment]
@SocketSecurity ignore-all

**Comment by nikoferro** [comment]
@tommasini thanks for reviewing! Just handled the security alerts, and also included screenshots of the swaps functionality working properly after the controller update. 

I have manually tested this PR myself, and the 4 scenarios described in the manual tests part of the description are working properly. Even in the case that there is something left untested, the E2E test that run post-merge also includes these 4 different scenarios.

As far as new features, this PR does not add any, it only adds support for Base swaps, but thats not a mobile feature per se. We are only adding the constants on the controller to enable swaps on that chain when we decide to enable the feature flag for this

**Comment by tommasini** [comment]
Seems amazing! Can just say why its okay to ignore all Security advices? https://github.com/MetaMask/metamask-mobile/pull/9584#issuecomment-2154849096

**Comment by nikoferro** [comment]
> Seems amazing! Can just say why its okay to ignore all Security advices? [#9584 (comment)](https://github.com/MetaMask/metamask-mobile/pull/9584#issuecomment-2154849096)

Ofc! Network accesses and Authors change have been reviewed already on the previous version of this PR (https://github.com/MetaMask/metamask-mobile/pull/9286)
 
The deprecated notice is something that we are carrying form previous `swaps-controller` version, and the work to solve these is still ongoing (you can check https://github.com/MetaMask/swaps-controller/issues)

As for the `web3` CVE, this is something thats from 2019 so we can assume it as safe, since its also marked as LOW/MILD, but `swaps-controller` actually does not include this version of `web3`. Part of this upgrade involves upgrading `web3` to its latest version (which is v4)

File: app/store/sagas/index.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ import Engine from '../../core/Engine';
 import Logger from '../../util/Logger';
 import LockManagerService from '../../core/LockManagerService';
 import AppConstants from '../../../app/core/AppConstants';
+import { XMLHttpRequest as _XMLHttpRequest } from 'xhr2';
+
+if (typeof global.XMLHttpRequest === 'undefined') {
+  global.XMLHttpRequest = _XMLHttpRequest;
```
[the remark this answers is not in the document]
Comment (nikoferro) [reply app/store/sagas/index.ts:21]: still necessary for now, i see you are working on https://github.com/MetaMask/metamask-mobile/pull/9910 so we might be able to remove this after that PR

**Pull request merged**